src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/config/WebClientConfiguration.kt

```diff
@@ -53,17 +42,23 @@ class WebClientConfiguration(
     return authorizedClientManager
   }
 
-  private fun addAuthHeaderFilterFunction() =
-    ExchangeFilterFunction { request: ClientRequest, next: ExchangeFunction ->
-      val token = when (val authentication = SecurityContextHolder.getContext().authentication) {
-        is AuthAwareAuthenticationToken -> authentication.token.tokenValue
-        else -> throw IllegalStateException("Auth token not present")
-      }
+  private fun getOauth2Client(authorizedClientManager: OAuth2AuthorizedClientManager, clientRegistrationId: String): ServletOAuth2AuthorizedClientExchangeFilterFunction {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId(clientRegistrationId)
+    return oauth2Client
+  }
 
-      next.exchange(
-        ClientRequest.from(request)
-          .header(HttpHeaders.AUTHORIZATION, "Bearer $token")
-          .build(),
-      )
-    }
+  private fun getExchangeStrategies(): ExchangeStrategies {
+    return ExchangeStrategies.builder()
+      .codecs { configurer: ClientCodecConfigurer -> configurer.defaultCodecs().maxInMemorySize(-1) }
```

Comment:
I see that we've done this in the past. By setting it to -1 we're setting it to unlimited. Can this negatively effect us at all ? More of a question than a request!

Reply:
No, not sure. Copied it from existing code base so not really sure about the settings used and how it impacts.

Reply:
All other repos use it as well. I won't block the PR but worth us figuring out what this actually means from a security perspective!